[UA] Disable reindexing for follower indices

sabarasaba opened the pull request:
Fixes: https://github.com/elastic/kibana/issues/215192

## Summary

This PR disables reindexing for follower indices and shows a callout recomending unfollowing the leader index.

Use the following [data.zip](https://github.com/user-attachments/files/19662819/data.zip) folder to test, should contain a follower index.


<details>
<summary>Screenshots</summary>

<img width="550" alt="Screenshot 2025-04-09 at 09 39 55" src="https://github.com/user-attachments/assets/7a0e6fa4-9d61-413a-b8fd-eafed4a911f1" />

![Screenshot 2025-04-11 at 10 03 34](https://github.com/user-attachments/assets/4cdec698-6504-4105-ba1d-64690e5407fe)
![Screenshot 2025-04-11 at 10 03 30](https://github.com/user-attachments/assets/a9acd7ee-bff7-4210-b701-f1e63d325e8a)


</details>

sabarasaba committed:
Conditionally disable reindex if its follower index

sabarasaba committed:
commit with @elastic email

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291175)
* Commit: 4f3395c472e070641b35a614263fbeb44d9b0050

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/291175#0196157d-90df-40ea-b258-01e9312a0969)

cc @sabarasaba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291175","number":291175,"commit":"4f3395c472e070641b35a614263fbeb44d9b0050"}],"number":291175}
buildkite-pr-comment-->

sabarasaba committed:
Fix broken tests and change copy

sabarasaba committed:
Add test

elasticmachine commented:
Pinging @elastic/kibana-management (Team:Kibana Management)

sabarasaba committed:
Add missing type

sabarasaba committed:
Fix snapshots

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291528)
* Commit: 8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d4c5144b8027f8688f95c87adfc46b59fc978832) | [after](https://github.com/elastic/kibana/commit/8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 214.1KB | 214.8KB | +660.0B |



### History
* :broken_heart: [Build #291482](https://buildkite.com/elastic/kibana-pull-request/builds/291482) failed 8841203db8d311bc592fbf54f76af9d87778eefc
* :broken_heart: [Build #291395](https://buildkite.com/elastic/kibana-pull-request/builds/291395) failed 348b34ef7c912fb48e39e017e28ed054cc6924e3
* :broken_heart: [Build #291175](https://buildkite.com/elastic/kibana-pull-request/builds/291175) failed 4f3395c472e070641b35a614263fbeb44d9b0050

cc @sabarasaba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291528","number":291528,"commit":"8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291482","number":291482,"commit":"8841203db8d311bc592fbf54f76af9d87778eefc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291395","number":291395,"commit":"348b34ef7c912fb48e39e017e28ed054cc6924e3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291175","number":291175,"commit":"4f3395c472e070641b35a614263fbeb44d9b0050"}],"number":291528}
buildkite-pr-comment-->

SoniaSanzV reviewed (commented):
Thank you for fixing this @sabarasaba! I've been doing some testing and I found weird that, after marking as read-only a follower index, the table and the flyout still suggest to reindex it, even though the button to doing so is not available.

<img width="1237" alt="Screenshot 2025-04-10 at 11 00 18" src="https://github.com/user-attachments/assets/3d75785e-5ed2-4bf3-ab1b-ddba6156d897" />

<img width="659" alt="Screenshot 2025-04-10 at 11 00 26" src="https://github.com/user-attachments/assets/54523694-0570-42e3-83b6-0063cf347a83" />

sabarasaba committed:
Fix small bug with resolution row and flyout contents

sabarasaba commented:
> Thank you for fixing this @sabarasaba! I've been doing some testing and I found weird that, after marking as read-only a follower index, the table and the flyout still suggest to reindex it, even though the button to doing so is not available.


Nice catch! I've addressed that with https://github.com/elastic/kibana/pull/217506/commits/74bd86a0da2f5e338d5c08345f3ababe954cf8f4 and updated the screenshots in the PR

sabarasaba commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch '8.x' into ua-follower_indices_fix

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292191)
* Commit: 2b01ced5ec488d9a67fbfe4462f0913476335b20

### Failed CI Steps
* [FTR Configs #62](https://buildkite.com/elastic/kibana-pull-request/builds/292191#01962403-10a1-4ae4-9516-689c37f142e3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/292191#01962403-10a1-4ae4-9516-689c37f142e3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/292191/jobs/01962403-10a1-4ae4-9516-689c37f142e3/artifacts/01962420-3215-4d05-ae2f-836cf1b56620) FTR Configs #<span></span>62 / saved objects tagging - functional tests visualize integration listing allows to filter by selecting a tag in the filter menu

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a5c0f3cb72a15223c659d54242f768d47440a6bc) | [after](https://github.com/elastic/kibana/commit/2b01ced5ec488d9a67fbfe4462f0913476335b20) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 214.0KB | 215.4KB | +1.4KB |



### History
* :green_heart: [Build #291528](https://buildkite.com/elastic/kibana-pull-request/builds/291528) succeeded 8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae
* :broken_heart: [Build #291482](https://buildkite.com/elastic/kibana-pull-request/builds/291482) failed 8841203db8d311bc592fbf54f76af9d87778eefc
* :broken_heart: [Build #291395](https://buildkite.com/elastic/kibana-pull-request/builds/291395) failed 348b34ef7c912fb48e39e017e28ed054cc6924e3
* :broken_heart: [Build #291175](https://buildkite.com/elastic/kibana-pull-request/builds/291175) failed 4f3395c472e070641b35a614263fbeb44d9b0050

cc @sabarasaba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292191","number":292191,"commit":"2b01ced5ec488d9a67fbfe4462f0913476335b20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291528","number":291528,"commit":"8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291482","number":291482,"commit":"8841203db8d311bc592fbf54f76af9d87778eefc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291395","number":291395,"commit":"348b34ef7c912fb48e39e017e28ed054cc6924e3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291175","number":291175,"commit":"4f3395c472e070641b35a614263fbeb44d9b0050"}],"number":292191}
buildkite-pr-comment-->

lcawl commented on x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx:
```diff
@@ -40,6 +40,30 @@ import { IndexClosedParagraph } from '../index_closed_paragraph';
+      title={
+        <FormattedMessage
+          id="xpack.upgradeAssistant.esDeprecations.indices.indexFlyout.detailsStep.followerIndexTitle"
+          defaultMessage="Unfollow leader index is recomended"
```
This seems to align with the following docs: https://www.elastic.co/guide/en/elasticsearch/reference/current/ccr-terminate-replication.html and https://staging-website.elastic.co/docs/deploy-manage/tools/cross-cluster-replication/ccr-terminate-replication, so I'd recommend similar terminology here:
```suggestion
          defaultMessage="Termination of replication is recommended"
```

lcawl commented on x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx:
```diff
@@ -40,6 +40,30 @@ import { IndexClosedParagraph } from '../index_closed_paragraph';
+      <p>
+        <FormattedMessage
+          id="xpack.upgradeAssistant.esDeprecations.indices.indexFlyout.detailsStep.followerIndexText"
+          defaultMessage="This index is a cross-cluster replication follower index. We recomend you to unfollow the leader index."
```
This paragraph alone wasn't sufficient for me to understand why I'd need to do this (and what the next steps were).  If we have room, i think it would be useful to convey more of the information from the issue description. For example:

```suggestion
          defaultMessage="This index is a cross-cluster replication follower index, which should not be reindexed. You can mark it as read-only or terminate the replication and convert it to a standard index."
```

In this case I think it would also be helpful to refer to the documentation about how to terminate replication. The aforementioned docs page doesn't exist in the doc link service yet, so it would needed to be added in https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-doc-links/src/get_doc_links.ts

lcawl commented on x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/resolution_table_cell.tsx:
```diff
@@ -67,12 +67,24 @@ const i18nTexts = {
+  reindexFollowerIndexTooltipLabel: i18n.translate(
+    'xpack.upgradeAssistant.esDeprecations.indices.reindexFollowerIndexTooltipLabel',
+    {
+      defaultMessage: 'Resolve this issue by unfollowing the leader index.',
```
>       defaultMessage: 'Resolve this issue by unfollowing the leader index.',

Per my other comments, since it seems like there's multiple steps involved (i.e. "pause the replication, close the index, unfollow, and reopen the index") I'd recommend using the terminology "terminate replication" to align with (and link to) the docs.

SoniaSanzV reviewed (approved):
thank you for addressing my feedback, tested locally and lgtm :)

jbudz commented:
@sabarasaba 

I updated the target branch from `old-8.x` (previously `8.x`) to `8.19` to  reflect our global branch rename of `8.x` to `8.19`.  

When you have a moment can you double check everything is in order and proceed as usual if so?  Feel free to reach out if there's any questions.

elasticmachine committed:
Merge branch '8.19' into ua-follower_indices_fix

sabarasaba committed:
Update x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

sabarasaba committed:
Update x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

sabarasaba committed:
Address CR

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/294552)
* Commit: 95e91d2b39ceba8d841aa93715d436524c6cb147

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d6b1c1f1617f7b028da6dc9d4fcacf3a7daa97ed) | [after](https://github.com/elastic/kibana/commit/95e91d2b39ceba8d841aa93715d436524c6cb147) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 215.2KB | 216.6KB | +1.5KB |



### History
* :yellow_heart: [Build #292191](https://buildkite.com/elastic/kibana-pull-request/builds/292191) was flaky 2b01ced5ec488d9a67fbfe4462f0913476335b20
* :green_heart: [Build #291528](https://buildkite.com/elastic/kibana-pull-request/builds/291528) succeeded 8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae
* :broken_heart: [Build #291482](https://buildkite.com/elastic/kibana-pull-request/builds/291482) failed 8841203db8d311bc592fbf54f76af9d87778eefc
* :broken_heart: [Build #291395](https://buildkite.com/elastic/kibana-pull-request/builds/291395) failed 348b34ef7c912fb48e39e017e28ed054cc6924e3
* :broken_heart: [Build #291175](https://buildkite.com/elastic/kibana-pull-request/builds/291175) failed 4f3395c472e070641b35a614263fbeb44d9b0050

cc @sabarasaba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/294552","number":294552,"commit":"95e91d2b39ceba8d841aa93715d436524c6cb147"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292191","number":292191,"commit":"2b01ced5ec488d9a67fbfe4462f0913476335b20"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291528","number":291528,"commit":"8ba08cc98a6fe0e8baaf1b1e8d6dc54a8c3209ae"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291482","number":291482,"commit":"8841203db8d311bc592fbf54f76af9d87778eefc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291395","number":291395,"commit":"348b34ef7c912fb48e39e017e28ed054cc6924e3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291175","number":291175,"commit":"4f3395c472e070641b35a614263fbeb44d9b0050"}],"number":294552}
buildkite-pr-comment-->
